### ISSUE TITLE
should help support --help? / help can't help help

### ISSUE THREAD

**Issue by djdv** [ipfs/go-ipfs#5557]
This is a strange one, but I'm reporting it anyway.
`ipfs help --help`
>Error: Unknown Command "help"

This seems to be the only command that doesn't support the `--help` flag.
While it probably wouldn't do anything more than the bare `help` subcommand itself, I found the error odd.

**Comment by kjzz** [comment]
So you meani that we should delete `ipfs help`?

**Comment by Stebalien** [comment]
Unfortunately, we actually use it. I said "leftover" because it looks like it may be a leftover from before we had a commands library (not sure, it may not have been).

However, we should probably make it behave better. The solution here is probably to just remove the length check. That is, convert "help" to "--help", even if there are other flags. Thoughts?

**Comment by djdv** [comment]
Would it be better to formalize it as a command? With this kind of alias mapping:
`ipfs help add` -> `ipfs add --help`?

**Comment by Stebalien** [comment]
Maybe? I'm not sure it's worth the effort (that would probably require deep changes to the commands library).

**Comment by kjzz** [comment]
> Would it be better to formalize it as a command? With this kind of alias mapping:
> `ipfs help add` -> `ipfs add --help`?

Agree,I think that ipfs <sub-command> --help maybe better,of course it will casue breanking?


> Unfortunately, we actually use it. I said "leftover" because it looks like it may be a leftover from before we had a commands library (not sure, it may not have been).

Sorry, i have misunderstand your reply.

**Comment by djdv** [comment]
For now, we could probably use the same hack, but append to args.
i.e.
remove arg[1]/"help" as well when detected.
`if helpcmd; args = append(args, "--help")`
in phase: `ipfs help add` -> `ipfs add` -> `ipfs add --help`

Right now, we alias `ipfs help` to `ipfs -h` but `ipfs --help` works the same, so it shouldn't break that case either.
It seems to work with other flags too.
`ipfs add -w --help` is the same as `ipfs add --help` alone.

And keep it in mind as the cmdslib evolves. See if it's necessary to formalize it as a `cmd` command.

Not sure if stacking hacks on hacks is a good idea, but for this specific command it seems harmless and unlikely to ever require change.
I guess the only thing we'd need to do is create some kind of stub `cmd` that is there only to hold the documentation for `help` and propagate it in `ipfs commands`.

Thoughts?

**Comment by kjzz** [comment]
Agree！@djdv , I am working for it right now.Thx a lot for advices.

**Comment by kjzz** [comment]
Hey @djdv  @Stebalien , i have create a pr #5573 about `ipfs help`.please help me review it.
If you think that it is the right way to fix it,i can continue to complete the test.Thx a lot.
